keel-api/src/test/kotlin/com/netflix/spinnaker/keel/integration/SchedulingResilienceTests.kt

```diff
@@ -0,0 +1,88 @@
+package com.netflix.spinnaker.keel.integration
+
+import com.jakewharton.retrofit2.adapter.kotlin.coroutines.CoroutineCallAdapterFactory
+import com.netflix.spinnaker.keel.KeelApplication
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import okhttp3.Interceptor
+import okhttp3.OkHttpClient
+import okhttp3.ResponseBody
+import okhttp3.mockwebserver.MockWebServer
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.boot.test.context.SpringBootTest.WebEnvironment.MOCK
+import org.springframework.context.annotation.Bean
+import org.springframework.test.context.junit.jupiter.SpringExtension
+import retrofit2.Retrofit
+import retrofit2.http.GET
+import strikt.api.Assertion
+import strikt.api.expectThat
+import java.io.IOException
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit.SECONDS
+
+// TODO: this doesn't really need to be an integration test except that it's painful to configure
+//       the retrofit client without Spring
+@ExtendWith(SpringExtension::class)
+@SpringBootTest(
+  classes = [KeelApplication::class, TestConfiguration::class],
+  webEnvironment = MOCK
+)
+internal class SchedulingResilienceTests {
+
+  @Autowired
+  lateinit var service: DummyRetrofitService
+
+  @Test
+  fun `retrofit call completes even if an interceptor throws an exception`() {
+    val latch = CountDownLatch(1)
+    GlobalScope.launch {
+      try {
+        service.greet().await()
+      } finally {
+        latch.countDown()
```

Comment:
Don't care if the method returns or throws an exception. Just ensuring it doesn't hang.